src/en/troubleshooting-upgrade.md

```diff
@@ -0,0 +1,48 @@
+Title: Juju troubleshooting - environment upgrade
+
+
+# Troubleshooting environment upgrades
+
+This section provides strategies and techniques to assist with broken
+environment upgrades. See
+[Upgrading Juju software](./juju-upgrade.html#upgrading-the-server-software) for
+information and instructions pertaining to upgrading your environment.
+
+
+## Case #1 - An agent does not restart (config)
+
+It may occur that an agent does not restart upon upgrade. One thing that may
+help is the inspection and modification of its `agent.conf` file. Comparing it
+with its file before upgrading can be very useful.
+
+Installing a different or modified configuration file will require a restart of
+the daemon. For example, for machine #2:
+
+```bash
+juju ssh 2 'ls -lh /etc/init/juju*'
+```
+
+```no-highlight
+-rw-r--r-- 1 root root 506 Sep  2 00:57 /etc/init/jujud-machine-2.conf
+-rw-r--r-- 1 root root 533 Sep  2 00:57 /etc/init/jujud-unit-mysql-0.conf
+```
+
+Therefore, if the agent for machine #2 is not coming up:
+
+```bash
+juju ssh 2
+<modify/restore /var/lib/juju/agents/machine-2/agent.conf>
```

Comment:
I am loathe to put this in a code block. Perhaps explain
" once you are running in the environment, you can modify or restore the `/var/lib/juju/agents/machine-2/agent.conf` file, then restart the agent with: